### ISSUE TITLE
Option for getting the LaTeX code

### ISSUE THREAD

**Issue by netw0rkf10w** [Khan/KaTeX#454]
MathJax has a contextual menu for that. It would be great to have such a feature in KaTeX.

**Comment by xymostech** [comment]
The LaTeX code for an equation is embedded in the MathML code, so you can extract it yourself if you want. This behaviour could maybe be put into an extension if you wanted it.

I don't think that the contextual menu is in the scope of KaTeX though, nor do I think we want it. Overriding the default right-click behaviour is usually just annoying. We also couldn't guarantee that it would work since KaTeX can be server-side rendered, but the menu would require some client-side script to run.

**Comment by netw0rkf10w** [comment]
@xymostech The contextual menu is just an example. If overriding the right-click is annoying, how about the left-click? Left-clicking the formula creates a popup window with LaTeX source code. I think that's not a bad idea.

**Comment by spicyj** [comment]
I don't think it is appropriate in most professional websites to expose the LaTeX code to the user. Technical users can use the devtools to see more info (as they would to inspect any other source code for the page) and if your site does want this functionality, it's probably best designed and integrated into your site in a way that KaTeX wouldn't be able to do and should be built custom for your site.

**Comment by netw0rkf10w** [comment]
@sophiebits @xymostech  Think about math forums. The equations are not there to be read only. For example, if somebody posts a question with a lot of equations, then for people who want to answer the question, it would be much more convenient for them to be able to copy the equations.

I am an administrator of a (big) math forum, and this is the most important reason why I'm still using MathJax (I once replaced it by KaTeX as an experiment, but that failed quickly). MathJax is very flexible: one can disable the option to view the source code, or can even disable completely the contextual menu.

**Comment by ylemkimon** [comment]
There is `copy-tex` extension, which puts LaTeX source code to the clipboard when rendered equation is copied. Also, in the extension, there is `katex2tex.js`, which extracts LaTeX source from rendered KaTeX element. You can use it to appropriate LaTeX source viewing method.

**Comment by edemaine** [comment]
I wrote `copy-tex`, and it would probably not be hard to write a similar extension that overrode some click behavior to reveal the source code.  I think it'd be fine to include such a feature as a KaTeX `contrib` extension, but agree that it doesn't belong in the "base" KaTeX (because that doesn't do any client-side JavaScript).  @netw0rkf10w are you interested in writing such an extension?  If not, I can give a try, though I'm not sure what the best behavior would be.

**Comment by netw0rkf10w** [comment]
@ylemkimon @edemaine  Awesome! I will definitely try it on my forum (not now but maybe in a month).

Thanks @edemaine for the great extension! The current "Select and copy" is already a very good option, but I think it would be nice to have an additional one that does not require selection (for example: right-click on the formula and "Copy LaTeX code", which I think is the best). Unfortunately I am currently not available to do it myself (although you have done almost all the work with `copy-tex`) as I'm busily writing my PhD thesis :(
